Update doc for using sync api with performance hints (#31121)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/running-inference/optimize-inference/high-level-performance-hints.rst

While an application is free to create more requests if needed (for example to support asynchronous inputs population) **it is very important to at least run the** ``ov::optimal_number_of_infer_requests`` **of the inference requests in parallel**. It is recommended for efficiency, or device utilization, reasons.

.. _working-with-sync-async-api:

Working with Sync/Async API
###########################

The API of the inference requests offers Sync and Async execution. The ``ov::InferRequest::infer()`` is inherently synchronous and simple to operate (as it serializes the execution flow in the current application thread). The Async "splits" the ``infer()`` into ``ov::InferRequest::start_async()`` and ``ov::InferRequest::wait()`` (or callbacks). For more information on synchronous and asynchronous modes, refer to the :doc:`OpenVINO Inference Request <../inference-request>`.

> If one application thread manage multiple infer requests, it is recommended to use the asynchronous (callbacks-based) API in production code. It is the most general and scalable way to implement the flow control for any possible number of requests. The ``THROUGHPUT`` and ``LATENCY`` performance hints automatically configure the Asynchronous pipeline to use the optimal number of processing streams and inference requests.

Reviewer: ```suggestion
If a single application thread manages multiple inference requests, the asynchronous (callbacks-based) API is recommended for production. It is the most scalable way to handle any number of requests. ``THROUGHPUT`` and ``LATENCY`` hints automatically configure the asynchronous pipeline for the optimal number of streams and requests.
```

> If a single application thread manages only one inference request, the synchronous API can be used with the `LATENCY` hint for single-thread use cases and the `THROUGHPUT` hint for multi-thread use cases. 

Reviewer: maybe something like?:

Use the ``LATENCY`` hint for single-threaded applications with one inference request. Use the ``THROUGHPUT`` hint for applications with multiple concurrent requests, such as in multi-threaded scenarios.
--------------------------------------------------------------------
If it doesn't cover the comment's point, and you want more details here, we can additionally mention something like 
---------------------------------------------------------------------
For a synchronous (blocking) call with one request, avoid using ``THROUGHPUT``, as it runs on a single stream that usually gets fewer resources due to even distribution across streams.

> If a single application thread manages only one inference request, the synchronous API can be used with the `LATENCY` hint for single-thread use cases and the `THROUGHPUT` hint for multi-thread use cases. 

Author: Yes, @maxnick. `Using the LATENCY hint for a synchronous inference (the blocking infer call for single infer request) with the best inference latency, and THROUGHPUT hint for applications with multiple concurrent infer requests in the fly with the best inference throughput.` is comment point of performance hint design.

Current document only introduced performance hint in async call scenario. This PR add performance hint in sync call scenario. Using sync call with `THROUGHPUT` hint in single-thread application is not reasonable.

> If a single application thread manages only one inference request, the synchronous API can be used with the `LATENCY` hint for single-thread use cases and the `THROUGHPUT` hint for multi-thread use cases. 

Reviewer 2: Ok, then can we rephrase the statement accordingly?

> If a single application thread manages only one inference request, the synchronous API can be used with the `LATENCY` hint for single-thread use cases and the `THROUGHPUT` hint for multi-thread use cases. 

Author: updated

> If one application thread only manage one infer request, the synchronous API can be used with `LATENCY` hint in single-thread use case and with `THROUGHPUT` hint in multi-threads use case. 

Reviewer: ```suggestion
If a single application thread manages only one inference request, the synchronous API can be used with the `LATENCY` hint for single-thread use cases and the `THROUGHPUT` hint for multi-thread use cases. 
```
